### PR TITLE
Closes #1584: Added dataset lock removal.

### LINKED ISSUE #1584: After remove version sent to review - contributor cannot create new one
1. User create new version dataset and send to review
2. Reviewer remove this version 
3. Info message still is displayed:   Submitted for Review – The draft version of this dataset is currently under review prior to publication. 
4. User cannot create new version - edit button are unavailable and see messege "  Submitted for Review – The draft version of this dataset is currently under review prior to publication.

### PR DESCRIPTION
**Pull request by MrK191**

### CODE REVIEW

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DeleteDatasetVersionCommand.java [hunk 1]
```diff
@@ -92,11 +94,26 @@ protected void executeImpl(CommandContext ctxt)  {
...
             }
 
             throw new IllegalCommandException("Cannot delete a released version", this);
         }
     }
+
+    /**
+     * Removes potential lock, otherwise the dataset would be permanently locked for user without ability to edit it.
+     */
+    private void RemoveLockAndUpdateDataset(CommandContext ctxt) {
```
Comment (madryk) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DeleteDatasetVersionCommand.java]: First letter of method should be lowercase.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DeleteDatasetVersionCommand.java [hunk 2]
```diff
@@ -92,11 +94,26 @@ protected void executeImpl(CommandContext ctxt)  {
...
+        doomed.getLocks().stream()
+              .filter(datasetLock -> datasetLock.getReason().equals(DatasetLock.Reason.InReview))
+              .findAny()
+              .ifPresent(datasetLock -> {
+                  ctxt.engine().submit(new RemoveLockCommand(getRequest(), doomed, DatasetLock.Reason.InReview));
+                  doomed.setModificationTime(new Timestamp(new Date().getTime()));
```
Comment (madryk) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DeleteDatasetVersionCommand.java:115]: Why we modify `modificationTime`?

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DeleteDatasetVersionCommand.java [hunk 3]
```diff
@@ -92,11 +94,26 @@ protected void executeImpl(CommandContext ctxt)  {
...
+
+    /**
+     * Removes potential lock, otherwise the dataset would be permanently locked for user without ability to edit it.
+     */
+    private void RemoveLockAndUpdateDataset(CommandContext ctxt) {
+        doomed.getLocks().stream()
+              .filter(datasetLock -> datasetLock.getReason().equals(DatasetLock.Reason.InReview))
+              .findAny()
+              .ifPresent(datasetLock -> {
+                  ctxt.engine().submit(new RemoveLockCommand(getRequest(), doomed, DatasetLock.Reason.InReview));
```
Comment (madryk) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DeleteDatasetVersionCommand.java]: I would go straight for `ctxt.datasets().removeDatasetLocks` method. Don't see why we should bother with executing separate command

**Review by madryk — CHANGES_REQUESTED** [review]

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DeleteDatasetVersionCommand.java (hunk 2 above)
Comment (MrK191) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DeleteDatasetVersionCommand.java:115]: Same thing is done in '~ReturnToAuthorCommand' so I wanted to stay consistent.

Comment (madryk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DeleteDatasetVersionCommand.java:115]: Ok, let's leave it. As far as I can see 'ModificationTime' on 'dvobject' has effect on creating sitemap, and deciding if we should reindex - so it should be fine

**Review by madryk — APPROVED** [review]

**Pull request merged**